### ISSUE TITLE
Warning: Couldn't write lextab module

### ISSUE THREAD

**Issue by git-afsantos** [timtadh/pyflwor#3]
Hello!

I'm trying your library, and so far it looks great. However, when compiling queries, I am getting some warnings, such as:

> WARNING: Couldn't write lextab module 'lextab'. [Errno 20] Not a directory: '/usr/local/lib/python2.7/dist-packages/pyflwor-1.2-py2.7.egg/pyflwor/lextab.py'

On one of my machines, I tried the `sudo pip install` method, which got me these warnings. I tried uninstalling, and going the `setup.py` method. It solved the issue on that machine, but on this other machine I still get the warnings. To be fair, I think the warning on my first machine had something to do with permissions, while this one says it cannot find the directory.

I am not using `virtualenv` in this case. Do you have any idea on how to solve this? The queries still work, it is just annoying to get the constant warnings.

**Comment by timtadh** [comment]
Hi @git-afsantos 

So this comes from the use of PLY under the hood as the parser generator. PLY one first invocation compiles and spits out a compiled version of the parser which it can use to save time on future invocations of the program. What is likely occurring is the way you are installing the library is preventing PLY from writing the file because it is in an unwritable directory and/or it is inside a zip file. 

I don't really know a good solution to this. You can probably use the warnings library to hide the warning.

**Comment by git-afsantos** [comment]
Thanks for the hint, I have solved the issue!

I uninstalled everything and started from scratch. Installed PLY on its own, and then installed from source using `pip` (this avoids installing as an egg, which was the issue -- cannot write a file within an egg).

Summing up:

```bash
cd pyflwor
sudo pip install ply
pip install . --user
```

Using the `--user` flag takes care of the permissions, and using `pip install .` avoids installing as an egg. Of course, all this would have been easier if I was using `virtualenv` from the start.
